beacon-chain/core/blocks/exit.go

```diff
@@ -50,37 +49,29 @@ func ProcessVoluntaryExits(
 	ctx context.Context,
 	beaconState state.BeaconState,
 	exits []*ethpb.SignedVoluntaryExit,
-) (state.BeaconState, error) {
+	exitInfo *v.ExitInfo,
+) (state.BeaconState, *v.ExitInfo, error) {
 	// Avoid calculating the epoch churn if no exits exist.
 	if len(exits) == 0 {
-		return beaconState, nil
+		return beaconState, exitInfo, nil
 	}
-	maxExitEpoch, churn := v.MaxExitEpochAndChurn(beaconState)
-	var exitEpoch primitives.Epoch
 	for idx, exit := range exits {
 		if exit == nil || exit.Exit == nil {
-			return nil, errors.New("nil voluntary exit in block body")
+			return nil, nil, errors.New("nil voluntary exit in block body")
 		}
 		val, err := beaconState.ValidatorAtIndexReadOnly(exit.Exit.ValidatorIndex)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		if err := VerifyExitAndSignature(val, beaconState, exit); err != nil {
-			return nil, errors.Wrapf(err, "could not verify exit %d", idx)
+			return nil, nil, errors.Wrapf(err, "could not verify exit %d", idx)
 		}
-		beaconState, exitEpoch, err = v.InitiateValidatorExit(ctx, beaconState, exit.Exit.ValidatorIndex, maxExitEpoch, churn)
-		if err == nil {
-			if exitEpoch > maxExitEpoch {
-				maxExitEpoch = exitEpoch
-				churn = 1
-			} else if exitEpoch == maxExitEpoch {
-				churn++
-			}
```

Comment:
This is not needed anymore because maximum exit epoch and churn are updated in the `ExitInfo` object